src/radical/pilot/agent/scheduler/base.py

```diff
@@ -762,36 +762,36 @@ def _unschedule_completed(self):
             self._refresh_ts_map()
 
 
-        for unit in to_unschedule:
-            # if we find a waiting unit with the same tuple size, we don't free
-            # the slots, but just pass them on unchanged to the waiting unit.
-            # Thus we replace the unscheduled unit on the same cores / GPUs
+        for task in to_unschedule:
+            # if we find a waiting task with the same tuple size, we don't free
+            # the slots, but just pass them on unchanged to the waiting task.
+            # Thus we replace the unscheduled task on the same cores / GPUs
             # immediately. This assumes that the `tuple_size` is good enough to
-            # judge the legality of the resources for the new target unit.
+            # judge the legality of the resources for the new target task.
 
-          # ts = tuple(unit['tuple_size'])
+          # ts = tuple(task['tuple_size'])
           # if self._ts_map.get(ts):
           #
           #     replace = self._waitpool[self._ts_map[ts].pop()]
-          #     replace['slots'] = unit['slots']
+          #     replace['slots'] = task['slots']
           #     placed.append(placed)
           #
-          #     # unschedule unit A and schedule unit B have the same
+          #     # unschedule task A and schedule task B have the same
           #     # timestamp
           #     ts = time.time()
-          #     self._prof.prof('unschedule_stop', uid=unit['uid'],
+          #     self._prof.prof('unschedule_stop', uid=task['uid'],
           #                     timestamp=ts)
           #     self._prof.prof('schedule_fast', uid=replace['uid'],
           #                     timestamp=ts)
           #     self.advance(replace, rps.AGENT_EXECUTING_PENDING,
           #                  publish=True, push=True)
           # else:
           #
-          #     # no replacement unit found: free the slots, and try to
-          #     # schedule other units of other sizes.
-          #     to_release.append(unit)
+          #     # no replacement task found: free the slots, and try to
+          #     # schedule other tasks of other sizes.
+          #     to_release.append(task)
 
-            to_release.append(unit)
+            to_release.append(task)
 
         if not to_release:
             if not to_unschedule:
```

Comment:
Is there a case where `to_release` is empty and `to_unschedule` is not?

Reply:
Out if scope.  But yes, there is, but that is the case where this code will not return at this point.